src/main/java/com/internet/shop/controllers/cart/CompleteOrderController.java

```diff
@@ -0,0 +1,35 @@
+package com.internet.shop.controllers.cart;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.Product;
+import com.internet.shop.service.OrderService;
+import com.internet.shop.service.ShoppingCartService;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CompleteOrderController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector INJECTOR = Injector.getInstance("com.internet.shop");
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) INJECTOR.getInstance(ShoppingCartService.class);
+    private final OrderService orderService =
+            (OrderService) INJECTOR.getInstance(OrderService.class);
+    private final UserService userService =
+            (UserService) INJECTOR.getInstance(UserService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Product> products = shoppingCartService.getByUserId(USER_ID).getProducts();
+        Order order = orderService.completeOrder(products, userService.get(USER_ID));
```

Comment:
let's change complete Order a little bit. As you can see You already have shoppingCart here but you will call your database once again for it inside completeOrder() method. So let's remove shoppingCartService.clear() from completeOrder() and use it here

Reply:
sure, thanks!)